### ISSUE TITLE
lastActionId is undefined

### ISSUE THREAD

**Issue by oscarbrown4** [ronniepettersson/homebridge-remootio#10]
Homekit shows no response.  Have disabled & re-enabled websocket api, restarted device multiple times, updated homebridge with new keys, etc.
Everytime I tap on the gate from homekit, this appears in the homebridge console: Unexpected error - lastActionId is undefined

**Comment by ronniepettersson** [comment]
@oscarbrown4 please provide the firmware version of the Remootio device, and the version of this module.  Enable debugging and provide the debug logs.

**Comment by oscarbrown4** [comment]
Remootio device version 2.3
homebridge-remootio v1.2.7

I disabled other homebridge plugins and restarted to make sure nothing interfered.



```[2/26/2022, 4:41:41 PM] [HB Supervisor] Restarting Homebridge...
[2/26/2022, 4:41:41 PM] [HB Supervisor] Starting Homebridge with extra flags: -I -D
[2/26/2022, 4:41:41 PM] [HB Supervisor] Started Homebridge v1.4.0 with PID: 4128
[2/26/2022, 4:41:41 PM] Loaded config.json with 10 accessories and 4 platforms.
[2/26/2022, 4:41:41 PM] Loaded 0 cached accessories from cachedAccessories.
[2/26/2022, 4:41:41 PM] ---
[2/26/2022, 4:41:41 PM] Loaded plugin: homebridge-config-ui-x@4.42.0
[2/26/2022, 4:41:41 PM] Registering platform 'homebridge-config-ui-x.config'
[2/26/2022, 4:41:41 PM] ---
[2/26/2022, 4:41:41 PM] Disabled plugin: homebridge-dummy@0.6.0
[2/26/2022, 4:41:41 PM] ---
[2/26/2022, 4:41:41 PM] Disabled plugin: homebridge-magic-occupancy@3.1.3
[2/26/2022, 4:41:41 PM] ---
[2/26/2022, 4:41:41 PM] Disabled plugin: homebridge-myq@2.7.4
[2/26/2022, 4:41:41 PM] ---
[2/26/2022, 4:41:42 PM] Loaded plugin: homebridge-remootio@1.2.7
[2/26/2022, 4:41:42 PM] Registering platform 'homebridge-remootio.Remootio'
[2/26/2022, 4:41:42 PM] ---
[2/26/2022, 4:41:42 PM] Disabled plugin: homebridge-ring@9.24.0
[2/26/2022, 4:41:42 PM] ---
[2/26/2022, 4:41:42 PM] Loading 4 platforms...
[2/26/2022, 4:41:42 PM] [Config] Initializing config platform...
[2/26/2022, 4:41:42 PM] [Config] Running in Service Mode
[2/26/2022, 4:41:42 PM] Ignoring config for the platform "myQ" in your config.json as the plugin "homebridge-myq" has been disabled.
[2/26/2022, 4:41:42 PM] Ignoring config for the platform "Ring" in your config.json as the plugin "homebridge-ring" has been disabled.
[2/26/2022, 4:41:42 PM] [Remootio] Initializing Remootio platform...
[2/26/2022, 4:41:42 PM] [Remootio] Initializing child bridge 0E:D0:EA:43:1C:CC
[2/26/2022, 4:41:42 PM] Loading 10 accessories...
[2/26/2022, 4:41:42 PM] Ignoring config for the accessory "MagicOccupancy" in your config.json as the plugin "homebridge-magic-occupancy" has been disabled.
[2/26/2022, 4:41:42 PM] Ignoring config for the accessory "MagicOccupancy" in your config.json as the plugin "homebridge-magic-occupancy" has been disabled.
[2/26/2022, 4:41:42 PM] Ignoring config for the accessory "DummySwitch" in your config.json as the plugin "homebridge-dummy" has been disabled.
[2/26/2022, 4:41:42 PM] Ignoring config for the accessory "DummySwitch" in your config.json as the plugin "homebridge-dummy" has been disabled.
[2/26/2022, 4:41:42 PM] Ignoring config for the accessory "DummySwitch" in your config.json as the plugin "homebridge-dummy" has been disabled.
[2/26/2022, 4:41:42 PM] Ignoring config for the accessory "DummySwitch" in your config.json as the plugin "homebridge-dummy" has been disabled.
[2/26/2022, 4:41:42 PM] Ignoring config for the accessory "DummySwitch" in your config.json as the plugin "homebridge-dummy" has been disabled.
[2/26/2022, 4:41:42 PM] Ignoring config for the accessory "DummySwitch" in your config.json as the plugin "homebridge-dummy" has been disabled.
[2/26/2022, 4:41:42 PM] Ignoring config for the accessory "DummySwitch" in your config.json as the plugin "homebridge-dummy" has been disabled.
[2/26/2022, 4:41:42 PM] Ignoring config for the accessory "DummySwitch" in your config.json as the plugin "homebridge-dummy" has been disabled.
[2/26/2022, 4:41:42 PM] Publishing bridge accessory (name: Homebridge 12B0, publishInfo: {
  username: '0E:6E:44:7A:12:B0',
  port: 51134,
  pincode: '***-**-***',
  category: 2,
  bind: undefined,
  mdns: undefined,
  addIdentifyingMaterial: true,
  advertiser: undefined
}).
Setup Payload:
X-HM://0024A1Y8L0TV6
Enter this code with your HomeKit app on your iOS device to pair with Homebridge:
                       
    ┌────────────┐     
    │ 489-13-813 │     
    └────────────┘     
                       
[2/26/2022, 4:41:42 PM] Homebridge v1.4.0 (HAP v0.10.0) (Homebridge 12B0) is running on port 51134.
[2/26/2022, 4:41:42 PM] [Remootio] Launched child bridge with PID 4132
[2/26/2022, 4:41:42 PM] Registering platform 'homebridge-remootio.Remootio'
[2/26/2022, 4:41:42 PM] [Remootio] Loaded homebridge-remootio v1.2.7 child bridge successfully
[2/26/2022, 4:41:42 PM] Loaded 1 cached accessories from cachedAccessories.0ED0EA431CCC.
[2/26/2022, 4:41:42 PM] [Remootio] Remootio platform finished initializing!
[2/26/2022, 4:41:42 PM] [Remootio] Configuring accessory Gate
[2/26/2022, 4:41:42 PM] Publishing bridge accessory (name: Remootio, publishInfo: {
  username: '0E:D0:EA:43:1C:CC',
  port: 55109,
  pincode: '***-**-***',
  category: 2,
  bind: undefined,
  mdns: undefined,
  addIdentifyingMaterial: true,
  advertiser: undefined
}).
[2/26/2022, 4:41:42 PM] [Remootio] Remotio platform didFinishLaunching()
[2/26/2022, 4:41:42 PM] [Remootio] IP: 192.168.50.109
[2/26/2022, 4:41:42 PM] [Remootio] SK: A0EFA28D25FA67C3D09C46AF44771027C56B2129573A43103EFB47AAEEC5F5D1
[2/26/2022, 4:41:42 PM] [Remootio] AK: 8C5F8E6F8BEE3C8150F517EB8BDFA754D8EE9C80E7B343D00ADC88941E4E2ACF
[2/26/2022, 4:41:42 PM] [Remootio] [Gate] Finished initializing!
[2/26/2022, 4:41:42 PM] [Remootio] [Gate] Connecting: attempt 1
[2/26/2022, 4:41:42 PM] Homebridge v1.4.0 (HAP v0.10.0) (Remootio) is running on port 55109.
[2/26/2022, 4:41:43 PM] [Remootio] [Gate] getCurrentStateHandler: Current door state: [Closed]
Unexpected error - lastActionId is undefined
[2/26/2022, 4:41:43 PM] [Remootio] [Gate] getTargetStateHandler: Target door state: [Closed]
[2/26/2022, 4:41:43 PM] [Remootio] [Gate] ObstructionDetected was requested
```

**Comment by ronniepettersson** [comment]
@oscarbrown4 , After further thinking about it, I believe your problem is that the API keys may be incorrect. you should have the following sequence in the log:

[2/26/2022, 6:50:16 PM] [Remootio] Initializing Remootio platform...
[2/26/2022, 6:50:16 PM] [Remootio] Remootio platform finished initializing!
[2/26/2022, 6:50:16 PM] [Remootio] Configuring accessory Garage
[2/26/2022, 6:50:16 PM] [Remootio] Remotio platform didFinishLaunching()
[2/26/2022, 6:50:16 PM] [Remootio] [Garage] Finished initializing!
[2/26/2022, 6:50:17 PM] [Remootio] [Garage] Connected
[2/26/2022, 6:50:18 PM] [Remootio] [Garage] Setting current state to CLOSED
[2/26/2022, 6:50:18 PM] [Remootio] [Garage] Authenticated

**Comment by ronniepettersson** [comment]
@oscarbrown4 try the new 1.2.10 release. I have improved the debugging from the underlying api library from Remootio.

**Comment by oscarbrown4** [comment]
Hi @ronniepettersson 
It seems it was an issue with my remootio device.  I copied and pasted the API keys, and turned off websocket API about a dozen times and it was still not working.  So I downloaded remootio's api client and tried to connect that way - no dice.  So I ended up completely resetting the remootio, getting new API keys, etc and now it works like a charm.
Apologies for wasting your time here, but thanks for responding quickly.  This is a great interface you made.  Shoot me your venmo, I'd love to buy you a beer.
Thanks!
